QStore4S/src/main/java/edu/asu/qstore4s/controller/QStore.java

```diff
@@ -199,32 +206,50 @@ public String getData(HttpServletRequest request, HttpServletResponse response,
     @ResponseBody
     @RequestMapping(value = "/query", method = RequestMethod.POST)
     public String query(HttpServletRequest request, HttpServletResponse response,
-            @RequestHeader("Accept") String accept, @RequestBody String query, @RequestParam("class") String clas)
-                    throws JSONException {
+            @RequestHeader("Accept") String accept, @RequestBody String query, @RequestParam("class") String clas,
+            @RequestParam("queryID") String queryID) throws JSONException, InterruptedException, ExecutionException {
```

Comment:
What if the caller provides a query id that is already in use? Qstore should create those ids itself and then return them to the caller for subsequent calls.